cmd/nom/main.go

```diff
@@ -13,17 +13,27 @@ import (
 )
 
 type Options struct {
-	Verbose    bool   `short:"v" long:"verbose" description:"Show verbose logging"`
-	Number     int    `short:"n" long:"number" description:"Number of results to show"`
-	Pager      string `short:"p" long:"pager" description:"Pager to use for longer output. Set to false for no pager"`
-	NoCache    bool   `long:"no-cache" description:"Do not use the cache"`
-	ConfigPath string `long:"config-path" description:"Location of config.yml"`
+	Verbose      bool     `short:"v" long:"verbose" description:"Show verbose logging"`
+	Number       int      `short:"n" long:"number" description:"Number of results to show"`
+	Pager        string   `short:"p" long:"pager" description:"Pager to use for longer output. Set to false for no pager"`
+	NoCache      bool     `long:"no-cache" description:"Do not use the cache"`
+	ConfigPath   string   `long:"config-path" description:"Location of config.yml"`
+	PreviewFeeds []string `short:"f" long:"feed" description:"Feed(s) URL(s) for preview"`
 }
 
 var ErrNotEnoughArgs = errors.New("not enough args")
 
 func run(args []string, opts Options) error {
-	cfg, err := config.New(opts.ConfigPath, opts.Pager, opts.NoCache)
+	if len(opts.PreviewFeeds) > 0 {
+		// Don't mess up the cache of configured feeds in the preview mode.
+		// Preview mode should use short-lived in-momory cache, but to make it
+		// happen we should support that through a cache interface.
+		// Another solution is to cache in different directory and wipe it out once nom is closed, i.e.
+		// defer wipeOutPreviewCache()
```

Comment:
Not sure we really need to cache preview feeds. By design they are previews and thus should pull the live, if users want to have content cached, the feeds should be permanent.

If you agree, we can remove this comment

Reply:
@guyfedwards Ahmmm, the problem with that is the fact that we do `fetchAllFeeds` on any `FindArticle` which does another HTTP request per feed, which is completely unnecessary since we are getting the same information we already fetched in the current session.

Let's say we are previewing 2 feeds:

- acme.com/rss.xml
- example.com/rss.xml

On the startup, we do 2 HTTP calls to get those feeds' contents.

Once user navigates to one of the articles, we do `FindArticle` which calls `fetchAllFeeds` resulting in 2 unnecessary HTTP calls. The more the user views articles, the more calls we are making in vain.

The repercussions are not only that the interface is sluggish, but we are risking requesting some of the feeds too often and getting rate-limited.

I've been playing around with that and here is what I came up with https://github.com/Nemoden/nom/commit/5c9c026e3199acd6f6679c205ffcd92d62f9d29a

But, I'm not sure how the design was envisioned and whether using caching mechanism is conceptually correct here.

Please let me know your thought around this.

Reply:
Ah I think I see the issue. The caching implementation was very basic and hidden behind fetch. Lets leave this comment in for now and can address teh caching for previews issue separately. Created https://github.com/guyfedwards/nom/issues/30 to track this